src/context/WalletProvider.tsx

```diff
@@ -362,6 +420,10 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
       return connectWC(chainId)
     }
 
+    if (name === WALLET_NAMES.Xdefi) {
+      return connectXdefi(chainId)
+    }
```

Comment:
Maybe also `goXdefi`?
```suggestion
    if (name === WALLET_NAMES.Xdefi) {
      if (!window.xfi) {
        goXdefi()
        return false
      }
      return connectXdefi(chainId)
    }
```
https://chrome.google.com/webstore/detail/xdefi-wallet/hmeobnfnfcmdkdcmlblgagmfpfboieaf